features/fixtures/mazerunner/jvm-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/Scenario.kt

```diff
@@ -29,26 +30,33 @@ abstract class Scenario(
     protected val eventMetadata: String?
 ) : Application.ActivityLifecycleCallbacks {
 
+    /**
+     * Tracks whether the scenario is starting Bugsnag only, or running the scenario.
+     */
+    protected var startBugsnagOnly: Boolean = false
+
     /**
      * Determines what log messages should be intercepted from Bugsnag and sent to Mazerunner
      * using a HTTP requests to the /logs endpoint. This is used to assert that Bugsnag is
      * behaving correctly in situations where sending an error/session payload is not
      * possible.
      */
-    open fun getInterceptedLogMessages() = emptyList<String>()
+    open fun getInterceptedLogMessages() = getZeroEventsLogMessages(startBugsnagOnly)
 
     /**
      * Initializes Bugsnag. It is possible to override this method if the scenario requires
      * it - e.g., if the config needs to be loaded from the manifest.
      */
-    open fun startBugsnag() {
+    open fun startBugsnag(startBugsnagOnly: Boolean) {
+        this.startBugsnagOnly = startBugsnagOnly
         Bugsnag.start(context, config)
     }
 
```

Comment:
I have another proposal here: rather than altering `Configuration` in the constructor or in `startBugsnag`, there should be a separate overriddable method called `prepareConfig(startBugsnagOnly: Boolean)`.

This would have a few advantages:

- Creates a consistent place for constructing configuration object (currently it's split between startBugsnag/init blocks)
- Allows for sensible defaults in `Scenario.kt` (such as setting `autoTrackSessions` to false)
- Reduces responsibilities of `startBugsnag`

Reply:
Yes, I think that would make sense - I'll raise a separate ticket to cover it, though, as it will end up touching quite a lot of files.

Reply:
PLAT-6049.